drivers/serial/uart_native_tty_bottom.c

```diff
@@ -0,0 +1,250 @@
+			cfsetispeed(ter, baudrate_lut[i].termios_baudrate);
+			return;
+		}
+	}
+	ERROR("Could not set baudrate, as %d is not supported.\n", baudrate);
+}
+
+/**
+ * @brief Set parity setting in the termios structure
+ *
+ * @param ter
+ * @param parity
+ */
+static inline void native_tty_baud_parity_set(struct termios *ter,
+					      enum native_tty_bottom_parity parity)
+{
+	switch (parity) {
+	case NTB_PARITY_NONE:
+		ter->c_cflag &= ~PARENB;
+		break;
+	case NTB_PARITY_ODD:
+		ter->c_cflag |= PARENB;
+		ter->c_cflag |= PARODD;
+		break;
+	case NTB_PARITY_EVEN:
+		ter->c_cflag |= PARENB;
+		ter->c_cflag &= ~PARODD;
+		break;
+	default:
+		/* Parity options mark and space are not supported on this driver. */
+		ERROR("Could not set parity.\n");
+	}
+}
+
+/**
+ * @brief Set the number of stop bits in the termios structure
+ *
+ * @param ter
+ * @param stop_bits
+ *
+ */
+static inline void native_tty_stop_bits_set(struct termios *ter,
+					    enum native_tty_bottom_stop_bits stop_bits)
+{
+	switch (stop_bits) {
+	case NTB_STOP_BITS_1:
+		ter->c_cflag &= ~CSTOPB;
+		break;
+	case NTB_STOP_BITS_2:
+		ter->c_cflag |= CSTOPB;
+		break;
+	default:
+		/* Anything else is not supported in termios. */
+		ERROR("Could not set number of data bits.\n");
+	}
+}
+
+/**
+ * @brief Set the number of data bits in the termios structure
+ *
+ * @param ter
+ * @param stop_bits
+ *
+ */
+static inline void native_tty_data_bits_set(struct termios *ter,
+					    enum native_tty_bottom_data_bits data_bits)
+{
+	unsigned int data_bits_to_set = CS5;
+
+	switch (data_bits) {
+	case NTB_DATA_BITS_5:
+		data_bits_to_set = CS5;
+		break;
+	case NTB_DATA_BITS_6:
+		data_bits_to_set = CS6;
+		break;
+	case NTB_DATA_BITS_7:
+		data_bits_to_set = CS7;
+		break;
+	case NTB_DATA_BITS_8:
+		data_bits_to_set = CS8;
+		break;
+	default:
+		/* Anything else is not supported in termios */
+		ERROR("Could not set number of data bits.\n");
+	}
+
+	/* Clear all bits that set the data size */
+	ter->c_cflag &= ~CSIZE;
+	ter->c_cflag |= data_bits_to_set;
+}
+
+int native_tty_open_tty_bottom(const char *pathname)
+{
+	int fd = open(pathname, O_RDWR | O_NOCTTY);
+
+	if (fd < 0) {
+		ERROR("Failed to open serial port %s, errno: %i\n", pathname, errno);
+	}
+
+	return fd;
+}
+
+int native_tty_configure_bottom(int fd, struct native_tty_bottom_cfg *cfg)
+{
+	int rc, err;
+	/* Structure used to control properties of a serial port */
+	struct termios ter;
+
+	/* Read current terminal driver settings */
+	rc = tcgetattr(fd, &ter);
+	if (rc) {
+		WARN("Could not read terminal driver settings\n");
+		return rc;
+	}
+
+	native_tty_termios_defaults_set(&ter);
+
+	native_tty_baud_speed_set(&ter, cfg->baudrate);
+	native_tty_baud_parity_set(&ter, cfg->parity);
+	native_tty_stop_bits_set(&ter, cfg->stop_bits);
+	native_tty_data_bits_set(&ter, cfg->data_bits);
+
+	if (cfg->flow_ctrl != NTB_FLOW_CTRL_NONE) {
+		WARN("Could not set flow control, any kind of hw flow control is not supported.\n");
```

Comment:
This warning can never be triggered when here.
Should this warning be better placed in native_tty_conv_to_bottom_cfg():104 ?